org.adempiere.ui/src/org/compiere/print/DrillReportCtl.java

```diff
@@ -0,0 +1,789 @@
+					if (rs.next())
+						defStr = rs.getString(1);
+					else {
+						if (log.isLoggable(Level.INFO))
+							log.log(Level.INFO, "(" + sPara.getColumnName() + ") - no Result: " + sql);
+					}
+				}
+				catch (SQLException e) {
+					log.log(Level.WARNING, "(" + sPara.getColumnName() + ") " + sql, e);
+				}
+				finally{
+					DB.close(rs, stmt);
+					rs = null;
+					stmt = null;
+				}
+			}
+			if (!Util.isEmpty(defStr))
+				value = defStr;
+		}	//	SQL Statement
+		else if (  variable.indexOf('@') != -1
+				&& variable.indexOf('@') != variable.lastIndexOf('@'))	//	we have a variable / BF [1926032]
+		{
+			//	Strip
+			int index = variable.indexOf('@');
+			String columnName = variable.substring(index+1);
+			index = columnName.indexOf('@');
+			if (index == -1)
+			{
+				log.warning(sPara.getColumnName()
+					+ " - cannot evaluate=" + variable);
+				return null;
+			}
+			String tail=index < (columnName.length()-1) ? columnName.substring(index+1) : null;
+			columnName = columnName.substring(0, index);
+			//	try Env
+			String env = Env.getContext(m_ctx, columnName);
+			if (env == null || env.length() == 0)
+				env = Env.getContext(m_ctx, columnName);
+			if (env.length() == 0)
+			{
+				log.warning(sPara.getColumnName()
+					+ " - not in environment =" + columnName
+					+ "(" + variable + ")");
+
+				if(DisplayType.isDate(sPara.getDisplayType())) {
+					return Env.getContext(m_ctx, "#Date");
+				}
+				return null;
+			}
+			else
+				value = env;
+
+
+
+			if (tail != null && columnName.equals("#Date"))
+			{
+				tail = tail.trim();
+				if (tail.startsWith("-") || tail.startsWith("+"))
+				{
+					boolean negate = tail.startsWith("-");
+					int type = Calendar.DATE;
+					tail = tail.substring(1);
+					if (tail.endsWith("d"))
+					{
+						tail = tail.substring(0, tail.length()-1);
+					}
+					else if (tail.endsWith("m"))
+					{
+						type = Calendar.MONTH;
+						tail = tail.substring(0, tail.length()-1);
+					}
+					else if (tail.endsWith("y"))
+					{
+						type = Calendar.YEAR;
+						tail = tail.substring(0, tail.length()-1);
+					}
+
+					int toApply = 0;
+					try
+					{
+						toApply = Integer.parseInt(tail);
+					} catch(Exception e){}
+					if (toApply > 0)
+					{
+						if (negate) toApply = toApply * -1;
+						Timestamp ts = toTimestamp(value);
+						Calendar cal = Calendar.getInstance();
+						cal.setTimeInMillis(ts.getTime());
+						cal.add(type, toApply);
+						value = new Timestamp(cal.getTimeInMillis());
+					}
+				}
+
+			}
+		}	//	@variable@
+		return value;
+	}
+
+	public String getTableName() {
+		return m_TableName;
+	}
+
+
+	public KeyNamePair[] getDrillProcessList() {
+		return drillProcessList;
+	}
+
+
+	public HashMap<Integer, KeyNamePair[]> getDrillTablePrintFormatMap() {
+		return drillTablePrintFormatMap;
+	}
+
+
+	public HashMap<Integer, KeyNamePair[]> getDrillProcessRulesPrintFormatMap() {
+		return drillProcessRulesPrintFormatMap;
+	}
+
+}
```

Comment:
again, two general rule: 1. javadoc for public constructor and method. 2. remove instead of comments out code.